peggy_on_the_hub.md

```diff
@@ -0,0 +1,45 @@
+# Peggy
+
+## Question
+
+Should Peggy be deployed as a Shared Security Zone or module in Gaia as part of the Cosmos SDK?
+
+## Context
+
+In April 2020, Althea proposed an [DeFi enabled Ethereum bridge for Cosmos](https://blog.althea.net/defi-enabled-ethereum-bridge-for-cosmos/) for a DEFI enable PegZone.
+
+The Interchain Foundation has provided [Althea](https://blog.althea.net/solid-foundations-for-peggy/)with resources to implement the technical components of this vision.
+
+Althea is developing a Cosmos SDK [module](https://github.com/cosmos/peggy/tree/althea-peggy)+ solidity contracts here.
+
+A Schelling point has not been established on how this module should be deployed.
+
+### Option 1
+
+ ATOM/(or another token:Althea?) holder set establishes a new blockchain purpose built for Peg zone management.  Via an on chain script powered by COSM-WASM, the Cosmos Hub adds the ability for validators to opt their ATOMs into faults on the Peggy chain and a COSM WASM contract is added that recognizes Peggy specific faults and slashes ATOMs. An additional property would all the victims to be reimbursed via slashed or inflated ATOMs from the Hub.
+
+### Option 2
+
+Deploy Peggy as module in Gaia as the Hub. Hub Validators ( maybe some day delegators) vote in the Liquidity DAO managing the collateral. ERC20 tokens are minted by the Hub onto the Cosmos Network. Interests rate/ earnings  would be distributed to staked ATOM holders.
+
+## The case for Option 1: Sovereign Chains with Shared Security
+
+In Option 1, the Cosmos Hub specializes as the security/ accountability layer of Cosmos. ATOM holders are not directly responsible for managing the Peg, the Ethereum event and instead a subset of validators opt into participation and governance of Peggy chain.
+
+Because of the separation of concerns between Peggy and the Cosmos Hub the Cosmos hub can credibly and independently respond to collusion and malfeasance by the operators of Peggy.
+
+The Cosmos Hub can offer this service to multiple similar Pegs rather than nominating a single blessed Peg zones.
+
+The Cosmos hub will be required to be able to analyze pairs of state transitions and smart contract executions on the Cosmos Hub for correctness. This will involve being able execute module logic and block headers and ethereal smart contract state. The most likely implementation target for these capabilities is Rust  executed inside CosmWasm.
+
+The Smart Contract can be made more efficient by a smaller number of signers.
```

Comment:
Depending on ETH gas prices this might be a deciding factor?

Reply:
The difference between 3 and 100 signers is 5x gas cost. So very sublinear thanks to our optimization work. Although I would like to do a much more detailed study before recommending any decisions.

Reply:
is the integration of althea-peggy into main branch of peggy not included in the job description from ICF?

I think the integrated full cycle of peggy branch is necessary for us to bring it to production level, although it can be done by others.

Reply:
this problem is more complex than it seems at first glance. We tried to address it more in this blog post.

https://blog.althea.net/solid-foundations-for-peggy/

Essentially the main branch of Peggy is targeting a level of security that requires additional research and theory work before it can be completed. In the process of doing this interesting and productive research they have left some components underdeveloped and others in a state where they can't be completed without more theory work.

Althea Peggy is targeting a lower level of security in exchange for having no open research questions, only implementation to complete.

This places everyone in a sort of awkward spot, Peggy master can't be put into production, but neither do we want to overwrite it.

Reply:
Thank you for the explanation! So we need a team to bring these two branches into an integrated production level peggy!